### ISSUE TITLE
env.block.random in queries returns a single byte of zero

### ISSUE THREAD

**Issue by luca992** [scrtlabs/SecretNetwork#1624]
In queries, `env.block.random` returns a `Vec<u8>[0]` (a byte array with a single byte which is zero). 

If randomness in queries is unsupported (which it appears it is), I think this behavior is dangerous and can lead to someone unknowing using the query block "random" as an actual random, when it is not random at all.

It would make much more sense to just return `None` in this case to completely avoid misuse.

**Comment by assafmo** [comment]
It is indeed not supported in queries since the source of randomness is the block header.

**Comment by luca992** [comment]
Yeah, understandable. Just returning `None` will prevent a lot of future easily avoidable problems then I think
